trackpy/artificial.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+from scipy.spatial import cKDTree
+from trackpy.utils import validate_tuple
+
+
+def draw_point(image, pos, value):
+    image[tuple(pos)] = value
+
+
+def feat_gauss(r, rg=0.333):
+    """ Gaussian at r = 0 with max value of 1. Its radius of gyration is
+    given by rg. """
+    return np.exp((r/rg)**2 * r.ndim/-2)
+
+
+def feat_gauss_edge(r, value_at_edge=0.1):
+    """ Gaussian at r = 0 with max value of 1. Its value at r = 1 is given by
+    value_at_edge. """
+    return np.exp(np.log(value_at_edge)*r**2)
+
+
+def feat_ring(r, r_at_max, value_at_edge=0.1):
+    """ Ring feature with a gaussian profile, centered at r_at_max. Its value
+    at r = 1 is given by value_at_edge."""
+    return np.exp(np.log(value_at_edge)*((r - r_at_max) / (1 - r_at_max))**2)
+
+
+def feat_hat(r, disc_size, value_at_edge=0.1):
+    """ Solid disc of size disc_size, with Gaussian smoothed borders. """
+    mask = r > disc_size
+    spot = (~mask).astype(r.dtype)
+    spot[mask] = feat_ring(r[mask], disc_size, value_at_edge)
+    spot[~mask] = 1
+    return spot
+
+
+def feat_step(r):
+    """ Solid disc. """
+    return r <= 1
+
+
+def draw_feature(image, position, diameter, max_value=None,
+                 feat_func=feat_gauss, ecc=None, **kwargs):
+    """ Draws a radial symmetric feature and adds it to the image at given
+    position.
+
+    Parameters
+    ----------
+    image : ndarray
+        image to draw features on
+    position : iterable
+        coordinates of feature position
+    diameter : number
+        defines the box that will be drawn on
+    max_value : number
+        maximum feature value. should be much less than the max value of the
+        image dtype, to avoid pixel wrapping at overlapping features
+    feat_func : function. Default: feat_gauss
+        function f(r) that takes an ndarray of radius values
+        and returns intensity values <= 1
+    ecc : positive number, optional
+        eccentricity of feature, defined only in 2D. Identical to setting
+        diameter to (diameter / (1 - ecc), diameter * (1 - ecc))
+    kwargs : keyword arguments are passed to feat_func
+    """
+    if len(position) != image.ndim:
+        raise ValueError("Number of position coordinates should match image"
+                         " dimensionality.")
+    diameter = validate_tuple(diameter, image.ndim)
+    if ecc is not None:
+        if len(diameter) != 2:
+            raise ValueError("Eccentricity is only defined in 2 dimensions")
+        if diameter[0] != diameter[1]:
+            raise ValueError("Diameter is already anisotropic; eccentricity is"
+                             " not defined.")
+        diameter = (diameter[0] / (1 - ecc), diameter[1] * (1 - ecc))
+    radius = tuple([(d - 1) / 2 for d in diameter])
```

Comment:
We use a continuous function and discretize it. A feature with diameter `d` has a radius `d / 2`, that is subsequently discretized according to the pixel grid. In the continuous world the `- 1` makes no sense.

I cannot explain my thought on this any better, does anybody see my point?